hi-ml-histopathology/src/histopathology/datamodules/panda_module.py

```diff
@@ -32,3 +34,63 @@ def get_splits(self) -> Tuple[PandaTilesDataset, PandaTilesDataset, PandaTilesDa
         return (PandaTilesDataset(self.root_path, dataset_df=splits.train),
                 PandaTilesDataset(self.root_path, dataset_df=splits.val),
                 PandaTilesDataset(self.root_path, dataset_df=splits.test))
+
+
+class SubPandaTilesDataModule(TilesDataModule):
+    """ subPandaTilesDataModule is the child class of PandaDataModule specific to PANDA dataset
+    Method get_splits() returns the train, val, test splits from a subset of the PANDA dataset specified by
+    train/validation dataframes.
+    """
+
+    def __init__(self, train_csv: Union[str, Path], val_csv: Union[str, Path], **kwargs: Any) -> None:
```

Comment:
We have a type annotation PathOrString for that. But generally, I would try to do without str for paths, and use Path throughout.